packages/client/tests/functional/issues/8832/tests.ts

```diff
@@ -0,0 +1,202 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+// https://github.com/prisma/prisma/issues/8832
+testMatrix.setupTestSuite(
+  ({ provider }) => {
+    async function createTags(length: number): Promise<number[]> {
+      const ids = Array.from({ length }, (_, i) => i + 1)
+      const data = ids.map((id) => ({ id }))
+      await prisma.tag.createMany({
+        data,
+      })
+      return ids
+    }
+
+    async function getTagsFindManyIn(ids: number[]): Promise<unknown[]> {
```

Comment:
```suggestion
    async function getTagsFindManyIn(ids: number[]): Promise<Tag[]> {
```

and import `Tag` above